src/components/ProviderInitializer/ProviderInitializer.tsx

```diff
@@ -52,34 +54,39 @@ import {
   refreshAccount
 } from 'utils/account';
 import { parseNavigationParams } from 'utils/parseNavigationParams';
-
 import { isContract } from 'utils/smartContracts';
+
 import {
   getOperaProvider,
   getCrossWindowProvider,
   getExtensionProvider,
   getPasskeyProvider,
   processModifiedAccount,
   getMetamaskProvider,
-  getIframeProvider
+  getIframeProvider,
+  handleGuardianWarning
 } from './helpers';
 import { useSetLedgerProvider } from './hooks';
 
 let initalizingLedger = false;
 
 export function ProviderInitializer() {
+  const { loginMethod, iframeLoginType } = useSelector(loginInfoSelector);
+
   const network = useSelector(networkSelector);
   const walletAddress = useSelector(walletAddressSelector);
   const walletConnectLogin = useSelector(walletConnectLoginSelector);
-  const { loginMethod, iframeLoginType } = useSelector(loginInfoSelector);
   const walletLogin = useSelector(walletLoginSelector);
   const address = useSelector(addressSelector);
   const ledgerAccount = useSelector(ledgerAccountSelector);
   const ledgerLogin = useSelector(ledgerLoginSelector);
   const isLoggedIn = useSelector(isLoggedInSelector);
   const chainID = useSelector(chainIDSelector);
   const tokenLogin = useSelector(tokenLoginSelector);
+  const userAccount = useSelector(accountSelector);
   const nativeAuthConfig = tokenLogin?.nativeAuthConfig;
+  const handleGuardianWarningToast = handleGuardianWarning();
```

Comment:
since it's not a hook, why does it export a function instead of being called directly in useEffect ?

Reply:
Updated. 👍